utils/zerovec/derive/src/ule.rs

```diff
@@ -0,0 +1,105 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use proc_macro2::Span;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+
+use syn::spanned::Spanned;
+use syn::{Data, DeriveInput, Error, Field, Ident};
+
+fn suffixed_ident(name: &str, suffix: usize, s: Span) -> Ident {
+    Ident::new(&format!("{name}_{suffix}"), s)
+}
+
+pub fn derive_impl(input: &DeriveInput) -> TokenStream2 {
+    if !crate::utils::has_valid_repr(&input.attrs) {
+        return Error::new(
+            input.span(),
+            "derive(ULE) must be applied to a #[repr(C)] or #[repr(transparent)] type",
+        )
+        .to_compile_error();
+    }
+    if input.generics.type_params().next().is_some()
+        || input.generics.lifetimes().next().is_some()
+        || input.generics.const_params().next().is_some()
+    {
+        return Error::new(
+            input.generics.span(),
+            "derive(ULE) must be applied to a struct without any generics",
+        )
+        .to_compile_error();
+    }
+    let struc = if let Data::Struct(ref s) = input.data {
+        if s.fields.iter().next().is_none() {
```

Comment:
Thought: I noticed that one of the `yoke` derives (not sure which) fails on this condition (it doesn't work on an empty struct).  Can we add this condition there, too?

Reply:
I think both of them should be fine with empty fields? Note that you can't even have a lifetime with zero fields so it always goes through the simpler lts=0 path.